### PR TITLE
docs: s3 backend update

### PR DESCRIPTION
**Pull request by stevenGravy**
Update s3 backend section to include that audit logs can be stored in s3 and refer to Athena.

### CODE REVIEW

**Review by ptgott — APPROVED** [review]

File: docs/pages/reference/backends.mdx [hunk 1]
```diff
@@ -573,8 +573,10 @@ teleport:
 
 ## S3
 
-S3 buckets can only be used as storage for the recorded sessions. S3 cannot
-store the audit log or the cluster state.
+S3 buckets can be used as storage for the recorded sessions and audit log.
```
Comment (zmb3) [inline docs/pages/reference/backends.mdx:576]: Suggestion: change the heading to `S3 (Session Recordings)` and use this wording intead:

Teleport supports using S3 as a backend for both session recordings
and audit logs. S3 cannot be used as the cluster state backend.
This section covers the use of S3 as a session recording backend.
For information on using S3 for audit logs, see the [Athena](#Athena)
section.

Comment (stevenGravy) [reply docs/pages/reference/backends.mdx]: thanks @zmb3 , please see update

**Review by scottpgallagher — APPROVED** [review]
Approved.

**Review by zmb3 — APPROVED** [review]
LGTM assuming you fix the DynamoDB lint failure.

**Pull request merged**